src/lib/modernExtend.ts

```diff
@@ -842,3 +937,162 @@ export function occupancy(args?: Partial<BinaryArgs>): ModernExtend {
+        'alarm_2': e.binary('alarm_2', ea.STATE, true, false).withDescription('Indicates whether IAS Zone alarm 2 is active'),
+        'tamper': e.binary('tamper', ea.STATE, true, false).withDescription('Indicates whether the device is tampered').withCategory('diagnostic'),
+        'battery_low': e.binary('battery_low', ea.STATE, true, false).withDescription('Indicates whether the battery of the device is almost empty')
+            .withCategory('diagnostic'),
+        'supervision_reports': e.binary('supervision_reports', ea.STATE, true, false)
+            .withDescription('Indicates whether the device issues reports on zone operational status')
+            .withCategory('diagnostic'),
+        'restore_reports': e.binary('restore_reports', ea.STATE, true, false)
+            .withDescription('Indicates whether the device issues reports on alarm no longer being present')
+            .withCategory('diagnostic'),
+        'ac_status': e.binary('ac_status', ea.STATE, true, false).withDescription('Indicates whether the device mains voltage supply is at fault')
+            .withCategory('diagnostic'),
+        'test': e.binary('test', ea.STATE, true, false).withDescription('Indicates whether the device is currently performing a test')
+            .withCategory('diagnostic'),
+        'battery_defect': e.binary('battery_defect', ea.STATE, true, false).withDescription('Indicates whether the device battery is defective')
+            .withCategory('diagnostic'),
+    };
+
+    const exposes: Expose[] = [];
+    const bothAlarms = args.zoneAttributes.includes('alarm_1') && (args.zoneAttributes.includes('alarm_2'));
+
+    let alarm1Name = 'alarm_1';
+    let alarm2Name = 'alarm_2';
+
+    if (args.zoneType === 'generic') {
+        args.zoneAttributes.map((attr) => exposes.push(exposeList[attr]));
+    } else {
+        if (bothAlarms) {
+            exposes.push(e.binary(args.zoneType + '_alarm_1', ea.STATE, true, false)
+                .withDescription(exposeList[args.zoneType].description + ' (alarm_1)'));
+            alarm1Name = args.zoneType + '_alarm_1';
+            exposes.push(e.binary(args.zoneType + '_alarm_2', ea.STATE, true, false)
+                .withDescription(exposeList[args.zoneType].description + ' (alarm_2)'));
+            alarm2Name = args.zoneType + '_alarm_2';
+        } else {
+            exposes.push(exposeList[args.zoneType]);
+            alarm1Name = args.zoneType;
+            alarm2Name = args.zoneType;
+        }
+        args.zoneAttributes.map((attr) => {
+            if (attr !== 'alarm_1' && attr !== 'alarm_2') exposes.push(exposeList[attr]);
+        });
+    }
+
+    const type = [];
+    if (args.msgType?.includes('notification')) type.push('commandStatusChangeNotification');
+    if (args.msgType?.includes('report')) type.push('attributeReport', 'readResponse');
+    if (type.length === 0) type.push('commandStatusChangeNotification');
```

Comment:
Let's include all converters by default, I don't see any reason not to.

Reply:
Fair enough, even tho the are not zcl compliant command types here